### PR TITLE
ktr lite house keeping

### LINKED ISSUE #441: KTRLite Enhancement
For enhancement of `KTRLite`:

- [x] unit-test
- [x] relabel
- [x] initializer 
- [x] revert to last version of knots plotting
- [x] tutorial update

### PR DESCRIPTION
**Pull request by edwinnglabs**
- [x] revert level knots plot
- [x] refine tutorial
- [x] refine some plotting code

## Description

It seems the older version of plot is more informative.  I'm proposing to revert it with some better implementation.

Fixes #441

### CODE REVIEW

File: orbit/models/ktrlite.py [hunk 1]
```diff
@@ -271,10 +282,13 @@ def _set_kernel_matrix(self, df):
             self.knots_tp_level = (1 + knots_idx_level) / self.num_of_observations
             self._level_knot_dates = df[self.date_col].values[knots_idx_level]
         else:
+            # to exclude dates which are not within training period
             self._level_knot_dates = pd.to_datetime([
-                x for x in self._level_knot_dates if 
-                (x <= df[self.date_col].max()) and (x >= df[self.date_col].min())
+                x for x in self._level_knot_dates if
+                (x <= df[self.date_col].values[-1]) and (x >= df[self.date_col].values[0])
```
Comment (wangzhishi) [inline orbit/models/ktrlite.py:288]: this is assuming that `df` is sorted by `date_col`, though this should be a requirement for our model.

Comment (wangzhishi) [reply orbit/models/ktrlite.py:288]: just curious if there is any special consideration in this change?

**Review by wangzhishi — APPROVED** [review]
LGTM!

File: orbit/models/ktrlite.py (hunk 1 above)
Comment (edwinnglabs) [reply orbit/models/ktrlite.py:288]: i have some impression that is what we check by default in the see` _validate_training_df` under base template.

**Pull request merged**